job-profile-wrangler/src/main/java/org/folio/graph/GraphReader.java

```diff
@@ -0,0 +1,219 @@
+
+  static {
+    // Configure the DOTImporter to create vertices based on the "name" attribute
+    DOT_IMPORTER.setVertexWithAttributesFactory((id, attrs) -> {
+      Map<String, String> normalAttrs = normalizeAttributes(attrs);
+      String name = normalAttrs.get("name");
+      switch (name) {
+        case "Job Profile" -> {
+          return JobProfileNode.fromAttributes(id, normalAttrs);
+        }
+        case "Match Profile" -> {
+          return MatchProfileNode.fromAttributes(id, normalAttrs);
+        }
+        case "Action Profile" -> {
+          return ActionProfileNode.fromAttributes(id, normalAttrs);
+        }
+        case "Mapping Profile" -> {
+          return MappingProfileNode.fromAttributes(id, normalAttrs);
+        }
+        default -> {
+          LOGGER.error("Unrecognized profile: '{}'", normalAttrs);
+          return null;
+        }
+      }
+    });
+
+    // Configure the DOTImporter to create edges based on the "label" attribute
+    DOT_IMPORTER.setEdgeWithAttributesFactory((attrs -> attrs.entrySet()
+      .stream()
+      .filter(entry -> entry.getKey().equals("label"))
+      .findFirst()
+      .map(entry -> {
+        Attribute attr = entry.getValue();
+        if (MatchRelationshipEdge.getLabelValue().equals(attr.getValue())) {
+          return new MatchRelationshipEdge();
+        } else if (NonMatchRelationshipEdge.getLabelValue().equals(attr.getValue())) {
+          return new NonMatchRelationshipEdge();
+        }
+        return null;
+      })
+      .orElse(new RegularEdge())
+    ));
+  }
+
+  private GraphReader() {}
+
+  /**
+   * Reads a graph from a DOT file specified by the repository path and ID.
+   *
+   * @param repoPath The path to the repository containing the DOT files.
+   * @param id       The ID of the graph to read.
+   * @return The read graph.
+   */
+  public static Graph<Profile, RegularEdge> read(String repoPath, int id) {
+    Path filePath = Paths.get(repoPath, GraphWriter.genGraphFileName(id));
+    Graph<Profile, RegularEdge> g = new SimpleDirectedGraph<>(RegularEdge.class);
+    DOT_IMPORTER.importGraph(g, filePath.toFile());
+    return g;
+  }
+
+  /**
+   * Reads all graphs from DOT files in the specified repository path.
+   *
+   * @param repoPath The path to the repository containing the DOT files.
+   * @return A list of all read graphs.
+   */
+  public static List<Graph<Profile, RegularEdge>> readAll(String repoPath) {
+    List<Graph<Profile, RegularEdge>> graphs = new ArrayList<>();
+    try (Stream<Path> stream = Files.list(Paths.get(repoPath))) {
+      stream
+        .filter(Files::isRegularFile)
+        .forEach(filePath -> {
+          Graph<Profile, RegularEdge> g = new SimpleDirectedGraph<>(RegularEdge.class);
+          DOT_IMPORTER.importGraph(g, filePath.toFile());
+          graphs.add(g);
+        });
+    } catch (IOException e) {
+      LOGGER.error("An error occurred while reading the directory.", e);
```

Comment:
@okolawole-ebsco  Could you please add here method name here, as we did the same in another place? Just to have consistent logs in all our modules.

Reply:
updated.